### PR TITLE
Add new formatting directives to solve OCaml idempotency issue

### PR DESCRIPTION
**Pull request by nbacquey**
This PR introduces two new formatting directives, which use custom scopes:
- `@append_scoped_multiline_delimiter` / `@prepend_scoped_multiline_delimiter`, which are the equivalent of `@*pend_multiline_delimiter`, but check if a custom scope is multi-line instead of the parent node.
- `@singleline_scoped_delete`, which is the equivalent of `@singleline_delete`, but checks if a custom scope is multi-line instead of the parent node.

These directives are used to fix the OCaml query file, which can now format
```ocaml
let not =
  function true -> false | false -> true
```
into
```ocaml
let not = function
  | true -> false
  | false -> true
```
in a single pass, instead of two previously.

Closes the OCaml part of #334:
```
cargo run -- -f tests/samples/input/ocaml.ml
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -820,6 +820,45 @@ This Tree-sitter query:
...
+```scheme
+; Delete the optional "|" before the first match case
+; if the context is single-line
+(function_expression
+  (match_case)? @do_nothing
+  .
+  "|" @singleline_scoped_delete
+  .
+  (match_case)
+  (#scope_id! "function_definiton")
```
Comment (torhovland) [inline README.md:838]:
```suggestion
  (#scope_id! "function_definition")
```

File: README.md [hunk 2]
```diff
@@ -820,6 +820,45 @@ This Tree-sitter query:
...
+#### Example
+
+```scheme
+; Add the optional "|" before the first match case
+; only if the context is multi-line
+(function_expression
+  "|"? @do_nothing
+  .
+  (match_case) @prepend_scoped_multiline_delimiter
+  (#scope_id! "function_definiton")
```
Comment (torhovland) [inline README.md:857]:
```suggestion
  (#scope_id! "function_definition")
```

File: languages/ocaml.scm [hunk 3]
```diff
@@ -789,17 +789,28 @@
   (#scope_id! "function_definiton")
 )
 (parenthesized_expression
-  (function_expression) @begin_scope @end_scope
+  (function_expression
+    "function" @append_spaced_scoped_softline
+  ) @begin_scope @end_scope
   (#scope_id! "function_definiton")
```
Comment (torhovland) [inline languages/ocaml.scm:795]:
```suggestion
  (#scope_id! "function_definition")
```

File: languages/ocaml.scm [hunk 4]
```diff
@@ -789,17 +789,28 @@
...
 )
 (function_expression
-  "|"* @do_nothing
+  (match_case)? @do_nothing
+  .
+  "|" @singleline_scoped_delete
   .
-  (match_case) @prepend_spaced_scoped_softline
+  (match_case)
   (#scope_id! "function_definiton")
```
Comment (torhovland) [inline languages/ocaml.scm:803]:
```suggestion
  (#scope_id! "function_definition")
```

File: languages/ocaml.scm [hunk 5]
```diff
@@ -789,17 +789,28 @@
...
 )
 (function_expression
-  "|"* @prepend_spaced_scoped_softline
+  "|"? @do_nothing
+  .
+  (match_case) @prepend_scoped_multiline_delimiter
+  (#scope_id! "function_definiton")
```
Comment (torhovland) [inline languages/ocaml.scm:809]:
```suggestion
  (#scope_id! "function_definition")
```

File: languages/ocaml.scm [hunk 6]
```diff
@@ -789,17 +789,28 @@
...
+  (#delimiter! "| ") ; sic
+)
+(function_expression
+  "|" @prepend_spaced_scoped_softline
   .
   (match_case)
   (#scope_id! "function_definiton")
```
Comment (torhovland) [inline languages/ocaml.scm:816]:
```suggestion
  (#scope_id! "function_definition")
```

File: languages/ocaml.scm [hunk 7]
```diff
@@ -789,17 +789,28 @@
...
```
Comment (torhovland) [inline languages/ocaml.scm:789]:
```suggestion
  (#scope_id! "function_definition")
```

File: src/lib.rs [hunk 8]
```diff
@@ -60,8 +60,15 @@ pub enum Atom {
     /// Represents a literal string, such as a semicolon.
     Literal(String),
     /// Represents a literal string, such as a semicolon. It will be printed only
-    // in multi-line constructs
+    // in multi-line node.
```
Comment (torhovland) [inline src/lib.rs:63]:
```suggestion
    // in multi-line nodes.
```

**Review by torhovland — APPROVED** [review]
Nice work!

Better do a global search for "definiton".

**Comment by nbacquey** [comment]
`definiton` is everywhere in the code, including files not related to OCaml. I think this warrants its own PR

File: src/lib.rs (hunk 8 above)
Comment (nbacquey) [reply src/lib.rs]: fixed in the new version

**Pull request merged**